### PR TITLE
feat: dapp permissions CRUD

### PR DESCRIPTION
**Pull request by emizzle**
Fixes: #95.

### CODE REVIEW

File: nim_status/lib/permissions.nim [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    dapp: Dapp
+    dappPermission: DappPermissions
+  var query = fmt"""SELECT {dapp.name.columnName}
+                    FROM {dapp.tableName}"""
+
+  let dapps = execQuery[Dapp](db, query)
+  if dapps.len == 0:
+    return result
+
+  # let tblDappPerms = initTable[string, DappPermissions]()
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:60]: :scissors:

File: nim_status/lib/permissions.nim [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  var tblDappPerms = dapps.map(dapp => (dapp.name, DappPermissions(name: dapp.name))).toOrderedTable()
+  # for dapp in dapps:
+  #   tb
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:63]: :scissors:

File: nim_status/lib/permissions.nim [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  permissions* {.
+    serializedFieldName("permissions"),
+    dbColumnName("permission")
+  .}: seq[string]
+
+
+proc addPermissions*(db: DbConn, dappPerms: DappPermissions) =
+  var dapp: Dapp
+  var dappPermission: DappPermissions
+  var query = fmt"""INSERT OR REPLACE INTO {dapp.tableName}
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:29]: Do you think there's a way to change this `var query` to a const?

File: nim_status/lib/permissions.nim [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+proc deletePermission*(db: DbConn, name: string) =
+  var dapp: Dapp
+  let query = fmt"""DELETE FROM
+                      {dapp.tableName}
+                    WHERE
+                      {dapp.name.columnName} = ?"""
+
+  db.exec(query, name)
+  # TODO: Should we also delete permissions from the permissions table?
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:90]: Probably not necessary due to the table having `FOREIGN KEY(dapp_name) REFERENCES dapps(name) ON DELETE CASCADE`

File: nim_status/lib/permissions.nim [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+                      {dappPermission.permissions.columnName}
+                    FROM {dappPermission.tableName}"""
+
+  let dappPerms = execQuery[DappPermissions](db, query)
+  for dappPerm in dappPerms:
+    if not tblDappPerms.hasKey(dappPerm.name):
+      continue
+    var existing = tblDappPerms[dappPerm.name]
+    ## existing.permissions = existing.permissions & dappPerm.permissions
+    # existing.permissions.insert(dappPerm.permissions, existing.permissions.len)
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:76]: :scissors:

File: nim_status/lib/permissions.nim [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    existing.permissions = existing.permissions & dappPerm.permissions
+    tblDappPerms[dappPerm.name] = existing
+  # dappPerms.map(dappPerm => tblDappPerms[dappPerm.name].insert dappPerm.permissions, tblDappPerms[dapp])
```
Comment (richard-ramos) [inline nim_status/lib/permissions.nim:79]: :scissors:

File: nim_status/lib/permissions.nim (hunk 3 above)
Comment (emizzle) [reply nim_status/lib/permissions.nim:29]: Its value is updated and reused later on in the proc, which is why it was a mutable `var`. Is there a reason it should be `const`?

**Pull request merged**